### PR TITLE
Add MNIST Redist UWP sample

### PR DESCRIPTION
**Pull request by tiagoshibata**

### CODE REVIEW

File: Samples/MNISTRedist/cs/MNIST_Demo.csproj [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+      <SubType>Designer</SubType>
+    </ApplicationDefinition>
+    <Page Include="MainPage.xaml">
+      <Generator>MSBuild:Compile</Generator>
+      <SubType>Designer</SubType>
+    </Page>
+  </ItemGroup>
+  <ItemGroup>
+    <PackageReference Include="Microsoft.AI.MachineLearning">
+      <Version>1.4.0-dev-20200715-0103-ee5ca27ae</Version>
```
Comment (tiagoshibata) [inline Samples/MNISTRedist/cs/MNIST_Demo.csproj:143]: Will be updated to the first Nuget package that has Store support

File: Samples/MNISTRedist/cs/MainPage.xaml.cs [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+using Windows.Storage.Streams;
+
+namespace MNIST_Demo
+{
+    public sealed partial class MainPage : Page
+    {
+        private mnistModel modelGen = null;
+        private mnistInput mnistInput = new mnistInput();
+        private mnistOutput mnistOutput;
+        //private LearningModelSession    session;
```
Comment (ryanlai2) [inline Samples/MNISTRedist/cs/MainPage.xaml.cs:21]: nit: commented out code

**Comment by ryanlai2** [comment]
Is it necessary to create a whole different project rather then modify the original mnist sample with optional preprocessor definitions to switch between inbox and redist versions of WinML? Squeezenet was changed to use the Redist in this style in this change : https://github.com/microsoft/Windows-Machine-Learning/commit/f850d3609225d88be3e30482c5a6e4f81e1f04d6

File: Samples/MNIST/UWP/cs/MNIST_Demo.csproj [hunk 3]
```diff
@@ -85,7 +85,7 @@
   </PropertyGroup>
   <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Release|x64'">
     <OutputPath>bin\x64\Release\</OutputPath>
-    <DefineConstants>TRACE;NETFX_CORE;WINDOWS_UWP</DefineConstants>
+    <DefineConstants>TRACE;NETFX_CORE;WINDOWS_UWP;USE_WINML_NUGET</DefineConstants>
```
Comment (ryanlai2) [inline Samples/MNIST/UWP/cs/MNIST_Demo.csproj:88]: Can you add the NuGet package reference ? That way, the MNIST sample will build without users needing to add the NuGet package themselves through VS NuGet package manager.

<PackageReference Include="Microsoft.AI.MachineLearning">
      <Version>1.4.0</Version>
</PackageReference>

image

**Comment by ryanlai2** [comment]
Can you add build configurations like "Debug_Inbox" "Debug_NuGet" "Release_Inbox" "Release_NuGet" that save USE_WINML_NUGET in the preprocessor definitions for the NuGet configuraitons?

# Save Build Configurations
Right-click on the solution in the Solution Explorer and select Properties. In the Properties window, click Configuration Manager. Click the drop-down menu for Active solution configuration and click <New...>. Enter the name of the new solution configuration and make sure Create new project configurations is checked. Now, preprocessor definitions can be saved in desired build configurations.

File: Samples/MNIST/UWP/cs/MNIST_Demo.csproj (hunk 3 above)
Comment (tiagoshibata) [reply Samples/MNIST/UWP/cs/MNIST_Demo.csproj:88]: Yes, I'll do it when the Nuget with store support is released.

**Review by ryanlai2 — APPROVED** [review]

**Pull request merged**